### PR TITLE
Technika

### PR DESCRIPTION
**Pull request by Gondolindrim**
Add Technika support

## Description

Adds support to Technika PCB.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/technika/config.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x8484 // TT for TealTechnik
+#define PRODUCT_ID      0x6049 // 60i
+#define DEVICE_VER      0x0001 // Revision pre-Alpha
+#define MANUFACTURER    TealTechnik
+#define PRODUCT         Technika
+#define DESCRIPTION	A custom PCB 45%
```
Comment (drashna) [inline keyboards/technika/config.h:26]:
```suggestion
```

File: keyboards/technika/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#define ____ KC_NO
```
Comment (drashna) [inline keyboards/technika/keymaps/default/keymap.c:17]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/technika/technika.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+bool led_update_kb(led_t led_state) {
+    if (led_update_user(led_state)) {
+        writePin(A15, led_state.num_lock);
+        writePin(B3, led_state.caps_lock);
+        writePin(B4, led_state.scroll_lock);
+    }   
+    return true;
+}
```
Comment (drashna) [inline keyboards/technika/technika.c:35]: https://docs.qmk.fm/#/feature_led_indicators?id=configuration-options

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/technika/rules.mk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality 
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/technika/rules.mk:19]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/technika/technika.c (hunk 3 above)
Comment (drashna) [reply keyboards/technika/technika.c:35]: Specifically, if you have the defines, you don't need the code, at all.

Comment (Gondolindrim) [reply keyboards/technika/technika.c:35]: The idea here is to have the LEDs programmable for both CAPS, NUM, LOCK or as layer indicators. As I can see from the indicator documentation you linked, I don't see how those defines allow for the LEDs to be configured as layer indicators, or do they?

Comment (fauxpark) [reply keyboards/technika/technika.c:35]: You can override the default behaviour by adding the `led_update_user()` function in your keymap, returning false, and using the defines in `layer_state_set_user()` instead.

File: keyboards/technika/keymaps/default/keymap.c (hunk 2 above)
Comment (Gondolindrim) [reply keyboards/technika/keymaps/default/keymap.c:17]: Committed in 144d22eae

File: keyboards/technika/technika.c (hunk 3 above)
Comment (Gondolindrim) [reply keyboards/technika/technika.c:35]: Nice ok guys committed in 66faf5b22

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**